server/src/test/java/io/crate/analyze/expressions/IntervalAnalysisTest.java

```diff
@@ -112,13 +126,34 @@ public void testIntervalInvalidStartEnd() throws Exception {
             .hasMessage("Startfield must be less significant than Endfield");
     }
 
+    @Test
+    public void testIntervalInvalidStartEndMillis() throws Exception {
+        assertThatThrownBy(() -> e.asSymbol("INTERVAL '1' MILLISECOND TO SECOND"))
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Startfield must be less significant than Endfield");
+    }
+
     @Test
     public void test_odd() throws Exception {
         var symbol = e.asSymbol("INTERVAL '100.123' SECOND");
         assertThat(symbol).isLiteral(new Period().withMinutes(1).withSeconds(40).withMillis(123)
                                          .withPeriodType(PeriodType.yearMonthDayTime()));
     }
 
+    @Test
+    public void test_odd_milliseconds() throws Exception {
+        var symbol = e.asSymbol("INTERVAL '101' MILLISECOND");
+        assertThat(symbol).isLiteral(new Period().withMillis(101)
+                                         .withPeriodType(PeriodType.yearMonthDayTime()));
+    }
+
+    @Test
+    public void test_odd_milliseconds_to_Second() throws Exception {
+        var symbol = e.asSymbol("INTERVAL '1.99' SECOND TO MILLISECOND");
+        assertThat(symbol).isLiteral(new Period().withSeconds(1).withMillis(990)
```

Comment:
naming: other way around: should be test_second_to_millisecond.

Reply:
I removed the `SECOND TO MILLISECOND` limiter because it is not in Postgres.